### ISSUE TITLE
Issue with "Symbol.iterator": …

### ISSUE THREAD

**Issue by tiborh** [mdn/content#668]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/iterator

#### What information was incorrect, unhelpful, or incomplete?
[...new Foo]
led to an error. (see below)

#### Specific section or headline?
User-defined iterables

#### What did you expect to see?
[ 1, 2, 3 ]

#### Did you test this? If so, how?
node script.js

```
SyntaxError: Unexpected token 'new'
    at wrapSafe (node:internal/modules/cjs/loader:1018:16)
    at Module._compile (node:internal/modules/cjs/loader:1066:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1131:10)
    at Module.load (node:internal/modules/cjs/loader:967:32)
    at Function.Module._load (node:internal/modules/cjs/loader:807:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47

```

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/symbol/iterator`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/iterator
* GitHub URL: https://github.com/mdn/content/blob/master/files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.html
* Report started: 2020-12-27T22:45:03.754Z

</details>

**Comment by tiborh** [comment]
My knowledge is not good enough to know what makes the difference, but I have now created two copies of the script.
one throws the above error, the other one does not.
[gives error]https://gitlab.com/tiborh/learning/-/blob/master/javascript/js/symbol_iterator_examples_error.js)
[runs good](https://gitlab.com/tiborh/learning/-/blob/master/javascript/js/symbol_iterator_examples_good.js)

**Comment by tiborh** [comment]
got the problem:
adding semicolons help:
`var myIterable = {}
myIterable[Symbol.iterator] = function* () {
    yield 1;
    yield 2;
    yield 3;
};
[...myIterable]; // [1, 2, 3]

class Foo {
  *[Symbol.iterator] () {
    yield 1;
    yield 2;
    yield 3;
  }
}

const someObj = {
  *[Symbol.iterator] () {
    yield 'a';
    yield 'b';
  }
};

[...new Foo]; // [ 1, 2, 3 ]
[...someObj]; // [ 'a', 'b' ]
`

**Comment by jpmedley** [comment]
Properly showing the results of iterators in a console as this example seems to intend requires a call to `Console.log()`. I've fixed this in #1040.

**Comment by jpmedley** [comment]
Closed via #1040
